xen/include/xen/vpci.h

```diff
@@ -335,6 +335,8 @@ static inline bool __must_check vpci_process_pending(struct vcpu *v)
 #ifdef CONFIG_HAS_VPCI_GUEST_SUPPORT
 bool vpci_translate_virtual_device(const struct domain *d, pci_sbdf_t *sbdf);
 #else
+#include <asm/pci.h>
```

Comment:
Wouldn't it be better to move the same line of code on line number 6 in this file above the condition `#ifdef CONFIG_HAS_VPCI`?

Reply:
sorry, I didn't pay attention that we use xen/pci.h instead of asm/pci.h there

Reply:
the problem is `#define is_hardware_pci_domain(d) (is_hardware_domain(d) && !hwdom_uses_vpci())`
few lines below. In particular hwdom_uses_vpci().

Potentially it just `false` if VPCI=n